### PR TITLE
Grades data and UI

### PR DESCRIPTION
**Pull request by ArjunK01**
# Status
In Progress

# What I did
Grades UI work, calling api for grades, rating, and hours data

## Screenshots
Screen Shot 2020-11-12 at 11 04 07 PM

### CODE REVIEW

**Comment by alex-shen1** [comment]
Small notes: don't forget to remove the big `Chart.bundle.js` file eventually because we have `npm` now. Also, it'd probably be good to rename your `chart.js` file to something like `grades_chart.js` for clarity.

File: tcf_website/models/models.py [hunk 1]
```diff
@@ -382,6 +430,78 @@ class Meta:
         ]
 
 
+class CourseGrade(models.Model):
+    course = models.ForeignKey(Course, on_delete=models.CASCADE, null=True)
+    subdepartment = models.CharField(max_length=255)
```
Comment (jackliu612) [inline tcf_website/models/models.py:435]: TODO: change this to a ForeignKey?

File: tcf_website/models/models.py [hunk 2]
```diff
@@ -382,6 +430,78 @@ class Meta:
...
+    total_enrolled = models.IntegerField(default=0)
+
+    def __str__(self):
+        return f"{self.subdepartment} {self.number} {self.average}"
+
+
+class CourseInstructorGrade(models.Model):
+    instructor = models.ForeignKey(
+        Instructor, on_delete=models.CASCADE, null=True)
+    first_name = models.CharField(max_length=225)
```
Comment (jackliu612) [inline tcf_website/models/models.py:468]: Are these needed if already instructor ForeignKey?

File: tcf_website/static/course/javascript/course_data.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
+
+let loadData = data => {
+  const { a_plus, a, a_minus, b_plus, b, b_minus, c_plus, c, c_minus, d_plus, d, d_minus, f, withdraw, drop } = data
+  const grades_data = [a_plus, a, a_minus, b_plus, b, b_minus, c_plus, c, c_minus, d_plus, d, d_minus, f, withdraw, drop]
+
+  console.log(data)
```
Comment (jackliu612) [inline tcf_website/static/course/javascript/course_data.js:6]: remove this

File: tcf_website/static/course/javascript/course_data.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+  createChart(grades_data)
+
+  exist(data.average_gpa) ? document.getElementsByClassName("gpa-text")[0].innerHTML = data.average_gpa + " GPA" : null
```
Comment (jackliu612) [inline tcf_website/static/course/javascript/course_data.js:10]: any way these could be looped? If not it's ok

File: tcf_website/templates/course/course_professor.html [hunk 5]
```diff
@@ -4,72 +4,198 @@
...
 
-    {% include "reviews/reviews.html" with reviews=reviews %}
+        {% include "reviews/reviews.html" with reviews=reviews %}
     </div>
 {% endblock %}
 
 {% block js %}
-<script type="module" src="{% static 'reviews/review.js' %}"></script>
-{% endblock %}
+    <script type="module" src="{% static 'reviews/review.js' %}"></script>
```
Comment (jackliu612) [inline tcf_website/templates/course/course_professor.html:190]: is this needed if chart.js is in npm package.json

File: tcf_website/models/models.py (hunk 2 above)
Comment (alex-shen1) [reply tcf_website/models/models.py:468]: yeah that's what i thought; it's part of the CSV data but i guess they made the script to do it

File: tcf_website/templates/course/course_professor.html (hunk 5 above)
Comment (alex-shen1) [reply tcf_website/templates/course/course_professor.html:190]: no idea

**Review by j-alicia-long — APPROVED** [review]

**Pull request merged**